WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit External Link/ProductExternalLinkViewController.swift

```diff
@@ -0,0 +1,232 @@
+        let title = NSLocalizedString("Product URL", comment: "")
+        let placeholder = NSLocalizedString("Enter URL", comment: "")
+        let viewModel = TitleAndTextFieldTableViewCell.ViewModel(title: title,
+                                                                 text: externalURL,
+                                                                 placeholder: placeholder,
+                                                                 keyboardType: .URL,
+                                                                 textFieldAlignment: .trailing,
+                                                                 onTextChange: { [weak self] text in
+                                                                    self?.externalURL = text
+        })
+        cell.configure(viewModel: viewModel)
+    }
+
+    func configureButtonText(cell: TitleAndTextFieldTableViewCell) {
+        let title = NSLocalizedString("Button Text", comment: "")
+        let placeholder = NSLocalizedString("Enter button text", comment: "")
+        let viewModel = TitleAndTextFieldTableViewCell.ViewModel(title: title,
+                                                                 text: buttonText,
+                                                                 placeholder: placeholder,
+                                                                 textFieldAlignment: .trailing,
+                                                                 onTextChange: { [weak self] text in
+                                                                    self?.buttonText = text ?? ""
+        })
+        cell.configure(viewModel: viewModel)
+    }
+}
+
+// MARK: - View Configuration
+//
+private extension ProductExternalLinkViewController {
+    func configureNavigation() {
+        title = NSLocalizedString("Product Link", comment: "Edit Product External Link navigation title")
+
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(completeEditing))
+    }
+
+    func configureMainView() {
+        view.backgroundColor = .listBackground
+    }
+
+    func configureTableView() {
+        tableView.dataSource = self
+
+        tableView.rowHeight = UITableView.automaticDimension
+        tableView.backgroundColor = .listBackground
+        tableView.removeLastCellSeparator()
+
+        view.addSubview(tableView)
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        view.pinSubviewToSafeArea(tableView)
+
+        registerTableViewCells()
+    }
+
+    func registerTableViewCells() {
+        tableView.register(TitleAndTextFieldTableViewCell.loadNib(), forCellReuseIdentifier: TitleAndTextFieldTableViewCell.reuseIdentifier)
+    }
+}
+
+// MARK: - Keyboard management
+//
+private extension ProductExternalLinkViewController {
+    /// Registers for all of the related Notifications
+    ///
+    func startListeningToNotifications() {
```

Comment:
I would not be opposed to using a longer but more specific name so when reading this inside `viewDidLoad`, it's much clearer what to expect. 

```suggestion
    func startObservingKeyboardNotifications() {
```

I'd also argue that it's probably not worth it to create a method for this since it's just one line. It probably makes the class easier to read since it's shorter? 🤷 Up to you though.

Reply:
I think it's still nice having a descriptive function even though the code is just one line. I like the name you suggested, updated in https://github.com/woocommerce/woocommerce-ios/pull/2442/commits/8a9971de59fadf9b837a9e21d92a724e23627d93